### PR TITLE
ci: Validate config-controller artifacts are generated

### PR DESCRIPTION
**Pull request by kylape**
Similar to the ACS operator, the config controller has code generation tasks that need to be run and the results checked in when making certain changes.  This adds a CI validation job using the established pattern to ensure these code generation task results are checked in as needed.

I've initially pushed a change to this branch that will intentionally cause the CI job to fail.  I'll then push the results of the code generation tasks and expect CI to pass.  Then of course I'll remove the dummy changes altogether and expect CI to still pass.

One difference from the ACS operator CI check is that I added a make target to the parent makefile instead of calling the config-controller makefile directly.  This is because I have a follow-up PR queued up that will copy a generated artifact to the Central Helm chart ([ROX-25493](https://issues.redhat.com/browse/ROX-25493)), and that command will go in the make target added in this PR.

### CODE REVIEW

**Comment by kylape** [comment]
Pasting test output here after the last commit:

```
Checking for diffs after making config-controller-gen...
diff --git a/config-controller/api/v1alpha1/zz_generated.deepcopy.go b/config-controller/api/v1alpha1/zz_generated.deepcopy.go
index 2f6d00c426..ca9eb44ba5 100644
--- a/config-controller/api/v1alpha1/zz_generated.deepcopy.go
+++ b/config-controller/api/v1alpha1/zz_generated.deepcopy.go
@@ -40,6 +40,21 @@ func (in *Deployment) DeepCopy() *Deployment {
 	return out
 }
 
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
+func (in *DummyStruct) DeepCopyInto(out *DummyStruct) {
+	*out = *in
+}
+
+// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new DummyStruct.
+func (in *DummyStruct) DeepCopy() *DummyStruct {
+	if in == nil {
+		return nil
+	}
+	out := new(DummyStruct)
+	in.DeepCopyInto(out)
+	return out
+}
+
 // DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
 func (in *Exclusion) DeepCopyInto(out *Exclusion) {
 	*out = *in
@@ -286,6 +301,7 @@ func (in *SecurityPolicySpec) DeepCopyInto(out *SecurityPolicySpec) {
 			(*in)[i].DeepCopyInto(&(*out)[i])
 		}
 	}
+	out.DummyField = in.DummyField
 }
 
 // DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new SecurityPolicySpec.
diff --git a/config-controller/config/crd/bases/config.stackrox.io_securitypolicies.yaml b/config-controller/config/crd/bases/config.stackrox.io_securitypolicies.yaml
index c4ee7b032e..0dec8d6eb3 100644
--- a/config-controller/config/crd/bases/config.stackrox.io_securitypolicies.yaml
+++ b/config-controller/config/crd/bases/config.stackrox.io_securitypolicies.yaml
@@ -49,6 +49,11 @@ spec:
                 type: string
               disabled:
                 type: boolean
+              dummyField:
+                properties:
+                  something:
+                    type: string
+                type: object
               enforcementActions:
                 items:
                   enum:
```

This is what I expected.  The test correctly identified the artifact changes and that they were not applied in the branch.

**Comment by kylape** [comment]
`check-generated-files` now passes :tada:

File: Makefile [hunk 1]
```diff
@@ -232,6 +232,11 @@ central-build-nodeps:
 	@echo "+ $@"
 	$(GOBUILD) central
 
+.PHONY: config-controller-gen
```
Comment (clickboo) [inline Makefile:235]: this should run with make gen on the top level as well so we will need to add this to the make generate target

Comment (kylape) [reply Makefile:235]: Can you explain what you mean by `make gen`? I tried both `make gen` and `make generate` at the top level, and it didn't work.  Are you suggesting we add that as a top-level make target?

Comment (clickboo) [reply Makefile:235]: I meant the top level repo make generated-srcs target should ensure controller stuff is also generated. So while having a separate target works, the ` go-generated-srcs` target should invoke this one too. Also there are clean targets for generated srcs which will need the corresponding change.

Comment (kylape) [reply Makefile]: Check out b04596f6d9ce10a7641a98ea8905d0c2b8bb9d7f and let me know if that's what you were thinking.

**Review by clickboo — APPROVED** [review]
LGTM, thanks for making the changes!

**Pull request merged**